packages/ab-testing/src/index.ts

```diff
@@ -1,9 +1,76 @@
-// add another package with `yo lerna-typescript:package @myscope/my-new-package` instead of `lerna create`
+import crc32 from 'fast-crc32c';
 
-export function greet(name: string): string {
-    return `ab-testing says: hello to ${name}`;
+type HashObject = (
+    object: { [key: string]: string | number | undefined },
+    salt: string
+) => { [s: string]: string };
+
+type ForceInclude = {
+    [hashedKey: string]: string[];
+};
+
+type Cohort = {
+    name: string;
+    allocation?: [number, number][];
+    force_include?: ForceInclude;
+};
+
+type Experiment = {
+    name: string;
+    cohorts: Cohort[];
+};
+
+export type ABTestingConfig = {
+    salt: string;
+    version: '1.0';
+    experiments: Experiment[];
+};
+
+export type UserProfile = {
+    user_id: number;
+    user_type?: string;
+    email?: string;
+    email_domain?: string;
+};
+
+function getModuloValue(experiment: string, userId: number): number {
+    return crc32.calculate(String(userId), crc32.calculate(experiment)) % 100;
 }
 
-export function add(a: number, b: number): number {
-    return a + b;
+export class Experiments {
+    config: ABTestingConfig;
+    profile: UserProfile;
+    hashedProfile: ReturnType<HashObject>;
+
+    constructor(config: ABTestingConfig, profile: UserProfile, hashProfile: HashObject) {
+        this.config = config;
+        this.profile = profile;
+        this.hashedProfile = hashProfile(profile, config.salt);
+    }
+
+    getCohort = (experimentName: string): string => {
+        const experimentConfig = this.config.experiments.find(e => e.name === experimentName);

```

Comment:
If the experiment isn't found, I think we should `console.error()` and bail out of this logic right away by `return 'control';`

This will reduce the amount of type check to do later on.